pkg/systemd/host.js

```diff
@@ -709,13 +755,107 @@ PageSystemInformationChangeSystime.prototype = {
+         *
+         * - systemd-timesyncd is actually available.
+         *
+         * The better alternative would be to have an API in
+         * o.fd.timedate1 for managing the list of NTP server
+         * candidates.
+         */
+
+        var checked = false;
+
+        $([ self.timedate1_service, self.timesyncd_service ]).on("changed", function () {
+            if (!checked &&
+                (self.timedate1_service.exists === false || self.timedate1_service.unit) &&
+                (self.timesyncd_service.exists !== null)) {
+
+                checked = true;
+
+                if (!self.timedate1_service.exists || self.timedate1_service.unit.Id !== "systemd-timedated.service") {
+                    console.log("systemd-timedated not in use, ntp server configuration not supported");
+                    return;
+                }
+
+                if (!self.timesyncd_service.exists) {
+                    console.log("systemd-timesyncd not available, ntp server configuration not supported");
+                    return;
+                }
+
+                if (!self.ntp_config_file)
+                    self.ntp_config_file = cockpit.file("/etc/systemd/timesyncd.conf.d/50-cockpit.conf",
+                                                        { superuser: "try" });
+
+                self.ntp_config_file.read().
+                    done(function (text) {
+                        if (text === null)
+                            self.ntp_servers = [ ];
+                        else {
+                            text.split("\n").forEach(function (line) {
+                                if (line.indexOf("NTP=") === 0)
+                                    self.ntp_servers = line.slice(4).split(" ").filter(function (val) {
+                                        return val !== "";
+                                    });
+                            });
+                        }
+                        if (self.ntp_servers.length === 0)
+                            self.ntp_servers = [ "" ];
+                        self.update_ntp_servers();
+                        self.update_ntp_status();
+                        self.update();
+                    }).
+                    fail(function (error) {
+                        console.log("error", error);
+                    });
+            }
+        });
+    },
+
+    set_ntp_servers: function() {
+        var self = this;
+
+        if (self.ntp_servers !== undefined) {
+            self.sync_ntp_servers();
+            var servers, text;
+            var promise;
+
+            servers = self.ntp_servers.filter(function (val) { return val !== ""; });
+            if (servers.length === 0)
+                text = null;
+            else
+                text = cockpit.format("[Time]\nNTP=$0\n", servers.join(" "));
+
+            return cockpit.spawn([ "mkdir", "-p", "/etc/systemd/timesyncd.conf.d" ], { superuser: "try" }).
```

Comment:
Should we have an option in file.replace() that does this? I see this starting to spread around in the code.